touchtechnology/common/tests/test_generic_views.py

```diff
@@ -207,3 +207,65 @@ def test_permissions_403(self):
             self.response_302()
 
         self.assertEqual(2, TestDateTimeField.objects.count())
+
+
+class GenericBulkCreateTests(TestCase):
+    def test_vanilla(self):
+        self.assertGoodView("generic:vanilla:bulk_create")
+        # The page should have a form for bulk creating multiple objects
+        self.assertResponseContains("<title>")  # Title will be set by template
+        
+        # Test creating multiple objects with bulk create
+        data = {
+            # management form hidden fields (default extra=3)
+            "form-TOTAL_FORMS": 3,
+            "form-INITIAL_FORMS": 0,
+            "form-MIN_NUM_FORMS": 0,
+            "form-MAX_NUM_FORMS": 1000,
+            "form-0-datetime_0": "2015-1-1",
+            "form-0-datetime_1": "9:00",
+            "form-0-datetime_2": "UTC",
+            "form-1-datetime_0": "2016-3-3",
+            "form-1-datetime_1": "10:00",
+            "form-1-datetime_2": "UTC",
+            # form-2 left empty to test partial submission
+        }
+        
+        res = self.post("generic:vanilla:bulk_create", data=data)
+        # Should create 2 objects (form-2 was empty)
+        self.assertEqual(2, TestDateTimeField.objects.count())
+        self.response_302()
+        self.assertRedirects(res, self.reverse("generic:vanilla:list"))
+
+        # Verify the objects were created correctly
+        objects = TestDateTimeField.objects.order_by("datetime")
+        self.assertEqual(str(objects[0].datetime), "2015-01-01 09:00:00+00:00")
+        self.assertEqual(str(objects[1].datetime), "2016-03-03 10:00:00+00:00")
```

Comment:
I would prefer using [`assertQuerysetEqual()`](https://docs.djangoproject.com/en/5.2/topics/testing/tools/#django.test.TransactionTestCase.assertQuerySetEqual) approach to testing the database state. Doing so would also eliminate the need to check the `count` on line 236 - it will be implicitly tested.

When using [`django-test-plus`](https://django-test-plus.readthedocs.io/en/latest/) we don't need to assign the `response` object, it will be implicitly be available as `self.last_response` so you can use that in the `assertRedirects` check.